app/helpers/application_helper.rb

```diff
@@ -26,6 +26,20 @@ def url_for_only_path(args)
     url_for(:only_path => true, **args)
   end
 
+  def api_resource_path(klass)
+    raise "API plugin not loaded" unless defined?(Api::ApiConfig)
+
+    identifier, _ = Api::ApiConfig.collections.find do |_, config|
+      next unless config.klass
+
+      klass <= config.klass.safe_constantize
+    end
+
+    raise "API route not found" unless identifier
+
+    send("api_#{identifier}_path")
```

Comment:
@abellotti What do you think about this approach? Seems pretty solid since it's driven by the API config.

Reply:
we have helpers for this. see https://github.com/ManageIQ/manageiq-api/blob/6096543dcfce5dd1ca1f6a8582cb7c834678a877/lib/api/collection_config.rb#L89 

I think something like identifier = Api::CollectionConfig.name_for_klass(klass) would do the trick

Reply:
Well, it should be more like:
```
identifier = Api::CollectionConfig.new.name_for_klass(klass)
```
which raises the same caching concerns as @gtanzillo mentioned [below](https://github.com/ManageIQ/manageiq-ui-classic/pull/7477#discussion_r516960364). Also this doesn't really work with STI :confused: when testing with `VmInfra`. Without STI support this method is not very useful :confused:

Reply:
Look at ```name_for_subclass(klass)`` in that file to see if that helps. In the API renderer we do ```name_for_klass(klass) || name_for_subclass(klass) ``` for some of the logic.

Reply:
:+1: